src/checkin/services/flow/step/RequestSeatsStep.ts

```diff
@@ -0,0 +1,32 @@
+import { Context } from '../../../model/Context'
+import StepTemplate from '../StepTemplate'
+
+export default class RequestSeatsStep extends StepTemplate {
+  when(context: Context): boolean {
+    const session = context.getSession()
+    return session.data.seats && session.data.seats.length === 0
+  }
+
+  onExecute(context: Context): Promise<boolean> {
+    const session = context.getSession()
+    if (!session.data.seats || session.data.seats.length === 0) {
```

Comment:
Maybe this boolean expression can be encapsulated in a function with a more intention revealing name so that the readability of this conditional can be improve